dist/ansible/scripts/ovn-setup.sh

```diff
@@ -10,23 +10,77 @@
 # This script is run as part of installation after the cluster is
 # up and before the ovn daemonsets are created.
 
-health=$(kubectl get --raw /healthz/ready)
-if [ $health != 'ok' ]
-then
-  echo " cluster not ready"
-  exit 1
-fi
+trap 'kill $(jobs -p); exit 0' TERM
+
+# Wait for cluster to come up
+wait_for_cluster_up () {
+  retries=0
+  while true; do
+    health=$(kubectl get --raw /healthz/ready)
+    if [[ $health != 'ok' ]]; then
+        (( retries += 1 ))
+        if [[ "${retries}" -gt 40 ]]; then
+          echo "error: cluster did not come up, exiting"
+          exit 1
+        fi
+        echo "info: Waiting for cluster to come up, waiting 2s ..."
+        sleep 2
+    else
+      if [[ "${retries}" != 0 ]]; then
+        echo "cluster came up in ${retries} 2sec tries"
+      fi
+      break
+    fi
+  done
+}
 
+# Wait for master-api to come up
+wait_for_api_server () {
+  retries=0
+  while true; do
+    api=$(kubectl get po -n kube-system -o wide | gawk '/master-api/{ print $7 }')
+    if [[ ${api} == '' ]]; then
+      (( retries += 1 ))
+      if [[ "${retries}" -gt 40 ]]; then
+        echo "error: master-api did not come up, exiting"
+        exit 1
+      fi
+      echo "info: Waiting for master-api to come up, waiting 2s ..."
+      sleep 2
+    else
+      if [[ "${retries}" != 0 ]]; then
+        echo "cluster came up in ${retries} 2sec tries"
+      fi
+      break
+    fi
+  done
+}
+
+#  ---------------
+wait_for_cluster_up
 
 # get api server
-api=$(kubectl get po -n kube-system -o wide | gawk '/master-api/{ print $7 }')
+wait_for_api_server
 apiserver=https://${api}:8443
 
+# from master-config.yaml
+#clusterNetworks:
+#  - cidr: 10.128.0.0/14
+#    hostSubnetLength: 9
+#    externalIPNetworkCIDRs:
+#  - 0.0.0.0/0
+#    networkPluginName: cni
+#  serviceNetworkCIDR: 172.30.0.0/16
+
 # get the network cidr (master-config.yaml clusterNetworkCIDR)
-net_cidr=10.128.0.0/14
+cfg=/etc/origin/master/master-config.yaml
```

Comment:
Can we take this path as an argument to the script (and just keep this as a default maybe) ?
Also, if one cannot obtain the cidr/sub values, can we resort to a default?

Reply:
@rajatchopra done. PTAL